source/javascripts/hooks/useStep.ts

```diff
@@ -0,0 +1,159 @@
+      icon: resolvedStepInfo?.info?.asset_urls?.['icon.svg'] || resolvedStepInfo?.info?.asset_urls?.['icon.png'],
+    };
+  }, [cvs, inputs, isLoadingInfo, isLoadingInputs, resolvedStepInfo, resolvedVersion, selectedVersion, versions]);
+};
+
+const useStepFromLocalApi = (cvs = ''): UseStepResult => {
+  const [id, version] = parseStepCVS(cvs);
+
+  let library: string = '';
+  if (isLocalStep(cvs)) {
+    library = 'path';
+  } else if (isGitStep(cvs)) {
+    library = 'git';
+  }
+
+  const { data, isLoading } = useQuery({
+    enabled: Boolean(id && library),
+    queryKey: ['/api/step-info', id, library, version],
+    queryFn: async ({ signal }) => {
+      const response = await fetch('/api/step-info', {
+        signal,
+        method: 'POST',
+        body: JSON.stringify({ id, library, version }),
+        headers: { 'Content-Type': 'application/json', Accept: 'application/json' },
+      });
+
+      return (await response.json()) as {
+        step?: Step;
+        info?: { asset_urls?: { [x: string]: string } };
+      };
+    },
+  });
+
+  return useMemo(() => {
+    return {
+      cvs,
+      isLoading,
+      step: data?.step,
+      selectedVersion: version,
+      resolvedVersion: version,
+      icon: data?.info?.asset_urls?.['icon.svg'] || data?.info?.asset_urls?.['icon.png'],
+    };
+  }, [cvs, isLoading, data, version]);
+};
+
+const useStep = (workflowId: string, stepIndex: number): UseStepResult | undefined => {
+  const { cvs, step: stepFromYml } = useStepFromYml(workflowId, stepIndex);
+
+  const stepFromAlgolia = useStepFromAlgolia(cvs);
+  const stepFromLocalApi = useStepFromLocalApi(cvs);
+
+  return useMemo(() => {
+    if (!cvs) {
+      return undefined;
+    }
+
+    const stepDefaults = stepFromAlgolia?.step || stepFromLocalApi?.step;
+
+    const inputs = stepDefaults?.inputs?.map(({ opts, ...input }) => {
+      const [inputName, defaultValue] = Object.entries(input)[0];
+
+      const inputFromYml = stepFromYml?.inputs?.find(({ opts: _, ...inputObjectFromYml }) => {
+        const inputNameFromYml = Object.keys(inputObjectFromYml)[0];
+        return inputNameFromYml === inputName;
+      });
+
+      return { opts, [inputName]: inputFromYml?.[inputName] ?? defaultValue };
+    });
+
+    const mergedStep = { ...merge({}, stepDefaults, stepFromYml), inputs };
+
+    return {
+      cvs,
+      step: mergedStep,
+      title: mergedStep.title || cvs,
+      maintainer: stepFromAlgolia?.maintainer,
+      availableVersions: stepFromAlgolia?.availableVersions,
+      isLoading: stepFromAlgolia?.isLoading || stepFromLocalApi?.isLoading,
+      selectedVersion: stepFromAlgolia?.selectedVersion || stepFromLocalApi?.selectedVersion,
+      resolvedVersion: stepFromAlgolia?.resolvedVersion || stepFromLocalApi?.resolvedVersion,
+      icon:
+        mergedStep?.asset_urls?.['icon.svg'] ||
+        mergedStep?.asset_urls?.['icon.png'] ||
+        stepFromAlgolia?.icon ||
```

Comment:
isnt the mergedStep already has everything merged to it, so the icons from the api and local api as well?